.pre-commit-config.yaml

```diff
@@ -0,0 +1,41 @@
+exclude: ^(img/)
+repos:
+- repo: https://github.com/pre-commit/pre-commit-hooks
+  rev: v4.6.0
+  hooks:
+    - id: check-added-large-files
+    - id: debug-statements
+    - id: detect-private-key
+    - id: end-of-file-fixer
+    - id: requirements-txt-fixer
+    - id: trailing-whitespace
+- repo: https://github.com/astral-sh/ruff-pre-commit
+  rev: v0.3.7
+  hooks:
+    - id: ruff
+      args: [--fix, --exit-non-zero-on-fix]
+- repo: https://github.com/psf/black
+  rev: 24.4.0
+  hooks:
+  - id: black
```

Comment:
question: why having both `ruff` and `black` ? Ruff apparently does what black does but faster. Can we rely on ruff only for the formatting?

Reply:
Yeah, we could.

I previously omitted using the ruff formatter since we already used the `black` as the formatter because I thought we could move forward one step at a time.

Reply:
Should we move with one and include the other only when we have a config for it?

Reply:
Suggestion: before pulling another dependency, let's decide which of the two we want. It can be your choice since you did the study for this. If we go with ruff, let's provide ruff config and remove black. If we stick to black, let's remove ruff as not needed.

Reply:
oh, ruff is currently used as a linter for now and black is used as a formatter. At the time when I did the transition in glauth operator, I did not fully trust the formatting in ruff yet since the version was low. So I thought we could move forward a bit with ruff as a linter to give it a try. But now probably I can discard black and use ruff to format the code. I can take a look at the ruff formatter.

And I forgot to add the config to `pyproject.toml` file like [this](https://github.com/canonical/glauth-utils/blob/main/pyproject.toml#L30-L49) for some reason. Will put it on.

Reply:
It's ok, we can address this in a subsequent PR, as long as we track this in an issue! Thanks